docs/src/lib/indexing.md

```diff
@@ -79,3 +79,19 @@ For performance reasons, accessing, via `getindex` or `view`, a single `row` and
 ## `setindex!`
 
 Under construction
+
+## Broadcasting
+
+`AbstractDataFrame` is converted to a matrix in broadcasting.
+`DataFrameRow`, `GroupedDataFrame`, `DataFrameRows` and `DataFrameColumns` are `collect`ed in broadcasting.
+
+Additionally it is possible to assign a value to `AbstractDataFrame` and `DataFrameRow` using the `.=` operator.
+In such an operation `AbstractDataFrame` is two dimensional and `DataFrameRow` as a single dimensional.
+
+If column indexing using symbols is performed the order of columns in the operation is specified by the order of symbols.
+
+It is allowed to perform `df[col] .= value` even if `col` is not present in the `DataFrame` and it is a `Symbol`.
+In such a case a new column will be created.
```

Comment:
This feels a little quirky to me.  I get that it could be handy, but it seems easier to introduce broadcasting first and then add this feature later.

Reply:
This was a major discussion point with @nalimilan. Without this feature all would be much simpler.

@nalimilan - do you think we have to stick with this feature? Of course the problem would be how to add a column of constant values to a data frame. You still could do `df.newcol = fill(value, nrow(df))` (or use the `Tables.allocatecolumn` thing), but initially you wanted to avoid having to do this.

Reply:
I guess we should summarize the discussion we had yesterday to give the rationale. The problem is that we currently allow `df[col] = value` to create a new column, so since we want to deprecate that to use `.=` for consistency with Base, we'd better provide a working replacement (or at least decide upfront whether we want it or not).

Reply:
Thank you @nalimilan for expanding on it. And we knew that having this would significantly add to the complexity of the design (as normally broadcasting assignment should not allocate new things (resize the target) and here we do this).

Reply:
Ah, I see.  Ok, yeah, I agree that this is definitely better than `df[:x] = 1` or even `df.x = 1`.  It's a shame we can't hook into `df.x .= 1`.  I guess folks aren't too thrilled with a two-step process like `addcol!(df, :x); df.x .= 1`?  Or I suppose an alternative spelling could be `df.x = Iterators.repeated(1)`, but I'm not familiar enough with DataFrame's  `setproperty!` conventions to know if this could happily coexist with the other meanings.

In any case, I'm sure you've already done thorough brainstorming here and this is a reasonable extension to make.  It's probably good to do this at the same time because it does impact the design rather substantially.

Reply:
Actually `df.x = Iterators.repeated(1)` (or something similar) is a possible option and an interesting idea.

In general our plan is to first do broadcasting and then redesign `setindex!` and `setproperty!` (the reason is that the former is new and the latter will have to go through deprecation and will be breaking).